src/HTSPDemuxer.cpp

```diff
@@ -226,6 +226,47 @@ PVR_ERROR CHTSPDemuxer::CurrentSignal ( PVR_SIGNAL_STATUS &sig )
   return PVR_ERROR_NO_ERROR;
 }
 
+int64_t CHTSPDemuxer::GetTimeshiftTime() const
```

Comment:
calling IsRealTimeStream ends up in grabbing the lock tree times. the results of the subtraction might be wrong too

````
bool CHTSPDemuxer::IsRealTimeStream() const
{
  if (GetTimeshiftTime() == 0)
    return true;

  if (GetTimeshiftBufferEnd() - GetTimeshiftTime() < 10)
    return true;

  return false;
}
```